haystack/components/evaluators/llm_evaluator.py

```diff
@@ -53,34 +53,87 @@ def __init__(
             Supported APIs: "openai".
         :param api_key:
             The API key.
-        :param examples:
-            Optional few-shot examples conforming to the expected input and output format as defined in the `inputs` and
-             `outputs` parameters.
-            Each example is a dictionary with keys "inputs" and "outputs"
-            They contain the input and output as dictionaries respectively.
+
         """
+        self.validate_init_parameters(inputs, outputs, examples)
+
         self.instructions = instructions
         self.inputs = inputs
-        self.outputs = outputs or ["score"]
+        self.outputs = outputs
+        self.examples = examples
         self.api = api
         self.api_key = api_key
-        self.examples = examples
-        expected_inputs = dict(inputs)
+
         if api == "openai":
             self.generator = OpenAIGenerator(api_key=api_key)
         else:
             raise ValueError(f"Unsupported API: {api}")
 
-        component.set_input_types(self, **expected_inputs)
+        template = self.prepare_template()
+        self.builder = PromptBuilder(template=template)
+
+        component.set_input_types(self, **dict(inputs))
+
+    def validate_init_parameters(
+        self, inputs: List[Tuple[str, Type[List]]], outputs: List[str], examples: List[Dict[str, Any]]
+    ):
+        """
+        Validate the init parameters.
+
+        :param inputs:
+            The inputs to validate.
+        :param outputs:
+            The outputs to validate.
+        :param examples:
+            The examples to validate.
+
+        :raises ValueError:
+            If the inputs are not a list of tuples with a string and a type of list.
+            If the outputs are not a list of strings.
+            If the examples are not a list of dictionaries.
+        """
+        # Validate inputs
+        if (
+            not isinstance(inputs, list)
+            or not all(isinstance(input, tuple) for input in inputs)
+            or not all(isinstance(input[0], str) and input[1] is not list and len(input) == 2 for input in inputs)
+        ):
+            msg = (
+                f"LLM evaluator expects inputs to be a list of tuples. Each tuple must contain an input name and "
+                f"type of list but received {inputs}."
+            )
+            raise ValueError(msg)
 
-    @component.output_types(results=List[List[Dict[str, Any]]])
+        # Validate outputs
+        if not isinstance(outputs, list) or not all(isinstance(output, str) for output in outputs):
+            msg = f"LLM evaluator expects outputs to be a list of str but received {outputs}."
+            raise ValueError(msg)
+
+        # Validate examples
+        if (
+            not isinstance(examples, list)
+            or not all(isinstance(example, dict) for example in examples)
+            or not all({"inputs", "outputs"} == example.keys() for example in examples)
```

Comment:
We can also check one level deeper, i.e., that both `inputs` and `outputs` are string dicts (`Dict[str, str]`)

Reply:
Okay done

Reply:
We now only enforce that the key is str though. value could still be a different type.

Reply:
For example, we want to support `List[str]` if there are multiple documents (their text contents) for one query.

Reply:
After internal discussion, we've decided not to restrict the value types to strings.